libyul/optimiser/DataFlowAnalyzer.cpp

```diff
@@ -117,27 +117,26 @@ void DataFlowAnalyzer::operator()(ForLoop& _for)
 	for (auto& statement: _for.pre.statements)
 		visit(statement);
 
+	AssignmentsSinceContinue assignmentsSinceCont;
+	assignmentsSinceCont(_for.body);
+
 	Assignments assignments;
 	assignments(_for.body);
 	assignments(_for.post);
 	clearValues(assignments.names());
 
 	visit(*_for.condition);
 	(*this)(_for.body);
+
+	if (!assignmentsSinceCont.empty())
+		clearValues(assignmentsSinceCont.names());
+
 	(*this)(_for.post);
 
 	clearValues(assignments.names());
-	popScope();
-}
+	// TODO: test: continue + assignment inside post{}
```

Comment:
Is that still pending?

Reply:
ah, you're right. I accidentally forgot to remove my personal note here.